lib/honeycomb/propagation/w3c.rb

```diff
@@ -42,14 +42,41 @@ def parse_v1(payload)
 
     # Serialize trace headers
     module MarshalTraceContext
-      def to_trace_header
+      TRACE_ID_REGEX = /^[A-Fa-f0-9]{32}$/.freeze
+      SPAN_ID_REGEX = /^[A-Fa-f0-9]{16}$/.freeze
+
+      def to_trace_header(propagation_context: nil)
+        if propagation_context.nil?
+          trace_id = trace.id
+          span_id = id
+        else
+          trace_id, span_id = propagation_context
+        end
         # do not propagate malformed ids
-        if trace.id =~ /^[A-Fa-f0-9]{32}$/ && id =~ /^[A-Fa-f0-9]{16}$/
-          return "00-#{trace.id}-#{id}-01"
+        if trace_id =~ TRACE_ID_REGEX && span_id =~ SPAN_ID_REGEX
+          return "00-#{trace_id}-#{span_id}-01"
         end
 
         nil
       end
     end
+
```

Comment:
I don't fully understand - if I am going to set a custom hook that does w3c propagation, how does one do that? There are no methods called `http_parser_hook` and `http_propagation_hook` like I would expect. In general, I don't fully understand why we have classes here at all.

Reply:
The configuration looks like this:
```
config.http_trace_parser_hook do |env|
    parser = Honeycomb::W3CPropagation::Parser.new
    parser.unmarshal_trace_context(env)
  end
```

but I just renamed some things to address this that I think will help. In the refactor above, I renamed `parse` to `unmarshal_trace_context` and renamed the above to `http_trace_parser_hook`. I think that should be more clear? So now the config will look like this (for a w3c parse):
```
  config.http_trace_parser_hook do |env|
    parser = Honeycomb::W3CPropagation::Parser.new
    parser.http_trace_parser_hook(env)
  end
```